### ISSUE TITLE
Configuration options cannot be parsed on Android in release builds

### ISSUE THREAD

**Issue by wwwdata** [pichillilorenzo/flutter_inappwebview#294]
After a quick debugging session for issue #237  I found out why it is not working. When Android is compiled for production, the code is optimized and named fields are renamed to a shorter variable name and mapped.

Unfortunately, the code for the configuration uses reflections to directly map values to field names. This cannot work in production when the optimizations are run because all the fields will have different names. It all works when you disable this feature, but this is not the solution. The solution would be to manually map everything without reflection features.

So release woks when minify and shrink is disabled in the release config
```
buildTypes {  
  release {       
    minifyEnabled false        
    shrinkResources false      
  }    
}
```

So, this code will fail for release when the variable names are optimized https://github.com/pichillilorenzo/flutter_inappwebview/blob/9c7ac0da8f01ceb40430da58397bdb31968ca5b1/android/src/main/java/com/pichillilorenzo/flutter_inappwebview/Options.java#L19-L26

I noticed that in release builds, lots of those errors were popping up. Example of an error message:

```
02-24 11:10:14.942 15395 15395 D InAppWebViewOptions: No field useShouldOverrideUrlLoading in class Lcom/pichillilorenzo/flutter_inappwebview/InAppWebView/H; (declaration of 'com.pichillilorenzo.flutter_inappwebview.InAppWebView.H' appears in base.apk)
02-24 11:10:14.942 15395 15395 D InAppWebViewOptions: No field preferredContentMode in class Lcom/pichillilorenzo/flutter_inappwebview/InAppWebView/H; (declaration of 'com.pichillilorenzo.flutter_inappwebview.InAppWebView.H' appears in ba
```

Essentially this means that on Android, it was always using the configuration defaults and it was not possible to override any configuration at all.

@pichillilorenzo are you already working on this for the big refactoring you are doing at the moment? Or are you aware of this issue? We basically just have to get rid of the reflection usage and then I guess a lot of Android bug reports can be closed. I guess the `shouldOverrideUrlLoading` is not the only issue related to this problem.

**Comment by wwwdata** [comment]
This is also related #280 
Sorry for not commenting there, but I guess here, we have the full explanation of why it is happening and can close the other issue.

**Comment by aboutmydreams** [comment]
thanks very much

**Comment by pichillilorenzo** [comment]
Released new version `3.0.0`!
